### PR TITLE
CRUD character backend + test

### PR DESCRIPTION
**Pull request by MicLamb**
CRUD on the backend of the character + tests.

Changed "forgeron" for "blacksmith".

#23

### CODE REVIEW

File: backend/test/integration/character.js [hunk 1]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
-    var Character = require('../../app/models/character');
-
-    Character.find({name:'testChar2'}).remove(function() {
-      var profile = {
-        "name" : "testChar2",
-        "nationality" : "Canadian",
-        "race" : "human",
-        "profession" : "miner",
-        "class" : "warrior",
-        "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 2]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
-      request(url)
-        .post('/character')
-        .send(profile)
+          var profile = {
+            "name" : "testCharacter",
+            "nationality" : "American",
+            "race" : "human",
+            "profession" : "professor",
+            "class" : "warrior",
+            "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 3]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+
+        it('should return error trying to create wrong class form character', function(done) {
+
+          var profile = {
+            "name" : "testCharacter",
+            "nationality" : "American",
+            "race" : "human",
+            "profession" : "miner",
+            "class" : "superhero",
+            "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 4]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+
+        it('should return error trying to create wrong skill form character', function(done) {
+
+          var profile = {
+            "name" : "testCharacter",
+            "nationality" : "American",
+            "race" : "human",
+            "profession" : "miner",
+            "class" : "superhero",
+            "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 5]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+          var Character = require('../../app/models/character');
+
+          Character.find({name:'testChar2'}).remove(function() {
+            var profile = {
+              "name" : "testChar2",
+              "nationality" : "Canadian",
+              "race" : "human",
+              "profession" : "miner",
+              "class" : "warrior",
+              "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 6]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+        it('should return error trying to create a character that already exists', function(done) {
 
-  it('should return error trying to create a character that already exists', function(done) {
+          var profile = {
+            "name" : "testChar2",
+            "nationality" : "Canadian",
+            "race" : "human",
+            "profession" : "miner",
+            "class" : "warrior",
+            "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 7]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+describe('Update', function() {
+  var characterId = null;
+  before(function(done) {
+    var character = new Character({
+      "name" : "testCharacter",
+      "nationality" : "American",
       "race" : "human",
       "profession" : "miner",
       "class" : "warrior",
       "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicism

File: backend/test/integration/character.js [hunk 8]
```diff
@@ -6,139 +6,427 @@ if(!mongoose.connection.readyState) {
...
+describe('Delete', function() {
+var characterId = null;
+  before(function(done) {
+    var character = new Character({
+      "name" : "testCharacter",
+      "nationality" : "American",
+      "race" : "human",
+      "profession" : "miner",
+      "class" : "warrior",
+      "belief" : "catholism",
```
Comment (Erikad94) [inline backend/test/integration/character.js]: catholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicismcatholicism

**Comment by DrPandemic** [comment]
:+1:

**Comment by Erikad94** [comment]
:+1:

**Comment by Baltazar510** [comment]
:+1:

**Pull request merged**